classify-split-extract-workflow/classify-job/split_and_classify.py

```diff
@@ -0,0 +1,358 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import re
+from google.cloud.documentai_toolbox import gcs_utilities
+from typing import Dict, Optional, List, Tuple
+
+from google.cloud import documentai_v1 as documentai
+from google.cloud import storage
+from google.cloud.documentai_v1.types.document import Document
+
+from pikepdf import Pdf
+from google.cloud.documentai_v1.types.document_processor_service import BatchProcessMetadata
+
+import config
+import docai_helper
+import gcs_helper
+import utils
+import bq_mlops
+from config import (
+    DOCAI_OUTPUT_BUCKET, NO_CLASSIFIER_LABEL, METADATA_CONFIDENCE,
+    METADATA_DOCUMENT_TYPE, SPLITTER_OUTPUT_DIR)
+from logging_handler import Logger
+
+storage_client = storage.Client()
+logger = Logger.get_logger(__file__)
+
+
+def batch_classification(
+        processor: documentai.types.processor.Processor,
+        dai_client: documentai.DocumentProcessorServiceClient,
+        input_uris: List[str]
+) -> Optional[Dict]:
+    """Performs batch classification on a list of documents using Document AI."""
+    logger.info(f"input_uris = {input_uris}")
+    if not input_uris:
+        return None
+
+    input_docs = [
+        documentai.GcsDocument(gcs_uri=doc_uri, mime_type=config.PDF_MIME_TYPE)
+        for doc_uri in input_uris
+    ]
+    gcs_documents = documentai.GcsDocuments(documents=input_docs)
+    input_config = documentai.BatchDocumentsInputConfig(gcs_documents=gcs_documents)
+
+    gcs_output_uri = f"gs://{DOCAI_OUTPUT_BUCKET}"
+    timestamp = utils.get_utc_timestamp()
+    gcs_output_uri_prefix = "classifier_out_" + timestamp
+    destination_uri = f"{gcs_output_uri}/{gcs_output_uri_prefix}/"
+    output_config = documentai.DocumentOutputConfig(
+        gcs_output_config={"gcs_uri": destination_uri}
```

Comment:
Use the classes wherever possible

Reply:
Could you elaborate please?

Reply:
Instead of using a dict for `gcs_output_config`, use the `documentai.GcsOutputConfig` class. Same for other request objects